src/foam/dao/CSVSink.js

```diff
@@ -40,22 +70,43 @@ foam.CLASS({
       name: 'isNewLine',
       value: true,
       visibility: 'HIDDEN'
+    },
+    {
+      name: 'axioms',
+      factory: function() {
+        return this.of.getAxiomByName('tableColumns').columns;
+      }
     }
   ],
 
   methods: [
-    function output(value) {
-      if ( ! this.isNewLine ) this.csv += ',';
-      this.isNewLine = false;
-      this.output_(value);
+    {
+      name: 'output',
+      args: [
+        { name: 'value' }
+      ],
+      code: function(value) {
+        if ( ! this.isNewLine ) this.csv += ',';
+        this.isNewLine = false;
+        this.output_(value);
+      },
+      javaCode: `
+        if ( ! getIsNewLine() ) setCsv(getCsv() + ",");
```

Comment:
We should use a StringBuilder in java rather than repeated string concatenation, it will be much faster.